tools/test/travis-ci/scancode_evaluate_test.py

```diff
@@ -40,67 +34,62 @@
  * limitations under the License.\
  */"
 
-
-# implement test class
-class TestScancodeEvaluate(TestCase):
-	""" Test scancode evaluation script """
-
-	def test_scancode_case_1(self):
-		""" Test Case 1 -- faulty json file
-			@inputs scancode_test_1.json
-			@outputs -1 if any error in file licenses found
-		"""
-		expected_result = -1
-		test_json = ROOT + "/scancode_test/scancode_test_1.json"
-
-		# pass json path to test function
-		result = license_check(ROOT, test_json)
-
-		self.assertEqual(expected_result, result)
-
-	def test_scancode_case_2(self):
-		""" Test Case 2 -- no errors in license headers, try multiple types i.e Apache-2.0, BSD3
-			@inputs scancode_test_2.json [4 Apache-2.0, 4 BSD-3.0]
-			@outputs 0
-		"""
-		expected_result = 0
-		test_json = ROOT + "/scancode_test/scancode_test_2.json"
-
-		result = license_check(ROOT, test_json)
-		self.assertEqual(expected_result, result, "False Negative(s)")
-
-	def test_scancode_case_3(self):
-		""" Test Case 3 -- all files containing errors
-			@inputs scancode_test_3.json [2 no header, 2 non-permissive + spdx, 1 missing SPDX]
-			@output 5
-		"""
-		# create stub files with a non-permissive license and missing spdx
-		for i in range(3, 6):
-			with open(stub_path + "test" + str(i) + ".h", "w") as file:
-				if i == 5:
-					file.write(invalid_header_2)
-				else:
-					file.write(invalid_header_1)
-
-		expected_result = 7
-		test_json = ROOT + "/scancode_test/scancode_test_3.json"
-
-		result = license_check(ROOT, test_json)
-
-		self.assertEqual(expected_result, result, "False Positive(s)")
-		# delete stub files
-		os.remove(stub_path + "test3.h")
-		os.remove(stub_path + "test4.h")
-		os.remove(stub_path + "test5.h")
-
-	def test_scancode_case_4(self):
-		""" Test Case 4 -- license header permissive and non-permissive 'license' [FP]
-			@inputs scancode_test_4.json
-			@outputs 0
-		"""
-		expected_result = 0
-		test_json = ROOT + "/scancode_test/scancode_test_4.json"
-
-		result = license_check(ROOT, test_json)
-
-		self.assertEqual(expected_result, result, "Non-Permissive Header False Positive")
+@pytest.fixture()
+def create_scanned_files():
+    """Create stub files.
+    test3.h missing license notice
+    test4.h with license notice
+    test5.h with license notice
+    """
+    file_paths = [
+        os.path.join(STUBS_PATH, "test3.h"),
+        os.path.join(STUBS_PATH, "test4.h"),
+        os.path.join(STUBS_PATH, "test5.h")
+    ]
+    for file_path in file_paths:
+        with open(file_path, "w") as new_file:
+            if file_path in [os.path.join(STUBS_PATH, "test3.h")]:
+                new_file.write(HEADER_WITHOUT_SPDX)
+            else:
+                new_file.write(HEADER_WITH_SPDX)
+    yield "create_scanned_files"
```

Comment:
I don't understand what this is for, isn't this fixture needed just to generate and tear down dependencies? Why yield a string?

Reply:
A remnant of a local experiment. It has now been removed.